src/lib/dnssd/minimal_mdns/core/tests/TestQName.cpp

```diff
@@ -40,299 +38,263 @@ static SerializedQNameIterator AsSerializedQName(const uint8_t (&v)[N])
     return SerializedQNameIterator(BytesRange(v, v + N - 1), v);
 }
 
-void IteratorTest(nlTestSuite * inSuite, void * inContext)
+
+TEST (TestQName, IteratorTest)
 {
     {
         static const uint8_t kOneItem[] = "\04test\00";
         SerializedQNameIterator it      = AsSerializedQName(kOneItem);
 
-        NL_TEST_ASSERT(inSuite, it.Next());
-        NL_TEST_ASSERT(inSuite, strcmp(it.Value(), "test") == 0);
-        NL_TEST_ASSERT(inSuite, !it.Next());
-        NL_TEST_ASSERT(inSuite, it.IsValid());
+        EXPECT_TRUE(it.Next());
+        EXPECT_FALSE(strcmp(it.Value(), "test"));
```

Comment:
although tehcnically true, this reeds ugly as "expect false it compared with test" when the actual test is "expect value to be test"). Prefer for strcmp to use "EXPECT_EQ(...., 0)" so that we are not pulled into reading the "FALSE" since it is not as fast to realize "false means they are equal".

we could maybe even create a macro or function so we can write `EXPECT_STREQUAL(it.Value(), "test")` which would be significantly more readable.

Reply:
https://pigweed.dev/pw_unit_test/#c.EXPECT_STREQ exists

Please update things to `EXPECT_STREQ(....)`